test/sql/parallelism/interquery/concurrent_attach_detach.cpp

```diff
@@ -55,54 +102,68 @@ struct DBInfo {
 	vector<TableInfo> tables;
 };
 
-DBInfo db_infos[db_count];
+void InterruptAttach(Connection &conn, const idx_t worker_id) {
+	AddLog(worker_id, "Interrupting connection");
+	conn.Interrupt();
+}
+
+DBInfo db_infos[DB_COUNT];
 
 class DBPoolMgr {
 public:
 	mutex mu;
 	map<idx_t, idx_t> m;
 
-	void addWorker(Connection &conn, const idx_t i) {
+	void AddWorker(Connection &conn, const idx_t db_id, const idx_t worker_id) {
 		lock_guard<mutex> lock(mu);
 
-		if (m.find(i) != m.end()) {
-			m[i]++;
-			return;
+		if (m.find(db_id) != m.end()) {
+			m[db_id]++;
+		} else {
+			m[db_id] = 1;
+		}
+
+		if (rand() % 2 == 0) {
+			// 50% chance to spawn an interruption worker
+			std::thread interruption(InterruptAttach, std::ref(conn), worker_id);
+			interruption.join();
 		}
-		m[i] = 1;
 
-		string query = "ATTACH '" + getDBPath(i) + "'";
-		execQuery(conn, query);
+		string query = "ATTACH IF NOT EXISTS'" + get_db_path(db_id) + "'";
+		ExecQuery(conn, query, worker_id);
```

Comment:
Hmm, I wonder if we need to spawn two workers here, one attaching and the other (spawned just after the attach worker) to interrupt it? At the moment, I am guessing that we're interrupting on the connection before the query starts (because of the `interruption.join()`)?

Also, maybe we should enable these on the main DB instance to ensure `ATTACH` takes some time:
```
SET checkpoint_threshold = '10.0 TB';
PRAGMA disable_checkpoint_on_shutdown;
```
Maybe that can be another (third) test, though. As we also want to test with checkpointing.